File: app/src/main/java/com/vikingsen/cheesedemo/ux/cheesedetail/CheeseDetailComponent.java
package com.vikingsen.cheesedemo.ux.cheesedetail;

import com.vikingsen.cheesedemo.inject.ActivityScope;

import dagger.Subcomponent;

@ActivityScope
@Subcomponent(modules = CheeseDetailModule.class)
public interface CheeseDetailComponent {
    void inject(CheeseDetailActivity target);
}

File: app/src/main/java/com/vikingsen/cheesedemo/ux/cheesedetail/CheeseDetailComponent.kt
package com.vikingsen.cheesedemo.ux.cheesedetail

import com.vikingsen.cheesedemo.inject.ActivityScope

import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = arrayOf(CheeseDetailModule::class))
interface CheeseDetailComponent {
    fun inject(target: CheeseDetailActivity)
}
